tests/gfid2path/gfid2path_fuse.t

```diff
@@ -7,13 +7,17 @@ CLI_SETGFID2PATH="gluster-setgfid2path";
 
 cleanup;
 
-XXHSUM_SOURCE="$(dirname $0)/../../contrib/xxhash/xxhsum.c $(dirname $0)/../../contrib/xxhash/xxhash.c"
-XXHSUM_EXEC=$(dirname $0)/xxhsum
-
-## Build xxhsum C source
-build_tester $XXHSUM_SOURCE -o $XXHSUM_EXEC -I$(dirname $0)/../../contrib/xxhash
-TEST [ -e $XXHSUM_EXEC ]
-
+if test -z $(type -p xxhsum); then
+    # Build xxhsum from contrib.
+    XXHSUM_SOURCE="$(dirname $0)/../../contrib/xxhash/xxhsum.c $(dirname $0)/../../contrib/xxhash/xxhash.c"
+    XXHSUM_EXEC=$(dirname $0)/xxhsum
+    build_tester $XXHSUM_SOURCE -o $XXHSUM_EXEC -I$(dirname $0)/../../contrib/xxhash
+else
+    # Use xxhsum found in PATH.
+    XXHSUM_EXEC=$(type -p xxhsum)
```

Comment:
@dmantipov just curious, why do you use `type -p xxhsum` instead of `which xxhsum`? Despite the disadvantage that type -p will not print anything if the command is aliased?

Reply:
This may be a bit of a personal taste writing shell scripts, but:

1. `type` is a shell builtin, and `which` is a shell function which executes an external program. On my system, it is:
```
$ type which
which is a function
which () 
{ 
    ( alias;
    eval ${which_declare} ) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot "$@"
}
```
2. This is **disadvantage** because an output of `which` should be parsed or otherwise handled in some way in case of error:
```
$ which nosuchcommand
/usr/bin/which: no nosuchcommand in (/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin)
```
OTOH zero output from `type -p nosuchcommand` clearly states that no `nosuchcommand` found in PATH.